docs/design/seccomp.md

```diff
@@ -0,0 +1,299 @@
+greater control over the security of workloads running in Kubernetes.
+
+Goals of this design:
+
+1.  Describe how to model seccomp profiles
+2.  Describe how to reference those profiles in the containers that use them
+
+## Constraints and Assumptions
+
+This design should:
+
+*  build upon previous security context work
+*  be container-runtime agnostic
+*  allow definition of custom profiles
+*  facilitate containerized applications that link directly to libseccomp
+
+## Use Cases
+
+1.  As an administrator, I want to be able to control access to seccomp profiles:
+    1.  To be able to default the seccmop profile when unspecified
+    2.  To be able to grant access to a seccomp profile to a class of users
+2.  As a user, I want to run an application with a seccomp profile similar to
+    the default one provided by my container runtime
+3.  As a user, I want to run an application which is already libseccomp-aware
+    in a container, and for my application to manage interacting with seccomp
+    unmediated by Kubernetes
+4.  As a user, I want to be able to describe a custom seccomp profile and use
+    it with my containers
+
+### Use Case: Administrator access control
+
+Controlling to control access to seccomp profiles is a cluster administrator concern.
+It should be possible for an administrator to:
+
+1.  Define a default seccomp profile for a class of users
+2.  Grant access to a seccomp profile to a class of users
+
+Administrators should be able to control two aspects of defaults for seccomp
+profiles:
+
+1.  Defaults for classes of users based on pod security policy
+2.  Defaults at the API level
+
+#### Classes of users
+
+The [pod security policy]() API extension governs the ability of users to make
+requests that affect pod and container security contexts.  The proposed design
+should deal with required changes to control access to new API features added.
+
+#### API Defaults
+
+There should be a minimum API-level guarantee for the seccomp profile to use
+when the field is not set when a pod is admitted to the API server.  This
+profile should be exposed in the control plane as an argument to the API
+server.
+
+### Use Case: Seccomp profiles similar to container runtime defaults
+
+Many users will want to use images that make assumptions about running in the
+context of their chosen container runtime.  Such images are likely to
+frequently assume that they are running in the context of the container
+runtime's default seccomp settings.  Therefore, it should be possible to
+express a seccomp profile similar to a container runtime's defaults.
+
+### Use Case: Applications that link to libseccomp
+
+Some applications already link to libseccomp and control seccomp directly.  It
+should be possible to run these applications unmodified in Kubernetes; this
+implies there should be a way to disable seccomp control in Kubernetes for
+certain containers, or to run with a "no-op" or "unconfined" profile.
```

Comment:
Just as a note, the cool thing with this is sometimes they can even use the default and then only restrict further from the app itself. So it's like seccomp++ :)